NewStarterProject/Services/AccessControlService.cs

```diff
@@ -0,0 +1,203 @@
+        {
+            var accessRecord =
+                await _context.AccessControlDoorsToUsers.SingleOrDefaultAsync(a =>
+                    a.AccessControlDoorsToUsersId == accessControlId);
+
+            if (accessRecord != null)
+            {
+                _context.AccessControlDoorsToUsers.Remove(accessRecord);
+                await _context.SaveChangesAsync();
+
+                return true;
+            }
+
+            return false;
+        }
+
+
+        /// <summary>
+        /// Returns a collection of Users who can access a door
+        /// </summary>
+        /// <param name="doorId"></param>
+        /// <returns></returns>
+        public async Task<List<UserDto>> GetUsersAssignedToDoor(int doorId)
+        {
+            return await _context.AccessControlDoorsToUsers.Include(a => a.User)
+                .Where(a => a.DoorId == doorId).Select(a => new UserDto()
+            {
+                UserId = a.User.UserId,
+                UserName = a.User.UserName ?? "",
+                FirstName = a.User.FirstName ?? "",
+                LastName = a.User.LastName
+            }).ToListAsync();
+        }
+
+
+        /// <summary>
+        /// Returns a collection of Users who can access a door
+        /// </summary>
+        /// <param name="doorId"></param>
+        /// <returns></returns>
+        public async Task<List<DoorDto>> GetDoorsAssignedToUser(int userId)
+        {
+            return await _context.AccessControlDoorsToUsers.Include(a => a.Door)
+                .Where(a => a.UserId == userId).Select(a => new DoorDto()
+                {
+                    DoorId = a.DoorId,
+                    DoorName = a.Door.DoorName
+                }).ToListAsync();
+        }
+
+
+
+        /// <summary>
+        /// Returns audit records - can be all records, or for a door or a user or a combination of both
+        /// </summary>
+        /// <param name="doorId"></param>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        public async Task<List<AuditRecordForDisplayDto>> GetAccessRecords(int? doorId, int? userId)
+        {
+            return await _context.AccessAudits
+                                 .Include(a => a.Door)
+                                 . Include(a => a.User)
+                                 .Where(a => (doorId == null || a.DoorId == doorId))
+                                 .Where(a => userId == null || a.UserId == userId)
+                                 .Select(a => new AuditRecordForDisplayDto()
+                                 {
+                                    DoorName = a.Door.DoorName,
+                                    UserName = a.User.UserName,
+                                    AuditDateTime = a.AuditDateTime,
+                                    AccessGranted = a.AccessGranted ? "Granted" : "Denied"
+                                 }).ToListAsync();
+        }
+
```

Comment:
This is pretty neat and I love the fact that you are returning a DTO specifically for displaying the Access Record. 

Unforuntately in P10 There are many places where we send back way too much info to the UI and the UI actually has to do a lot of processing on the information before displaying it to the user. 

This code here is great because we only send back the information required to display the UI, nice and simple! 😄